1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,49 @@
-function setAlarm() {}
+const display = document.querySelector('#timeRemaining');
+const secondsInput = document.querySelector('#alarmSet');
+const setButton = document.getElementById("set");
+const stopButton = document.getElementById('stop');
+
+let countdown;
+let totalSeconds;
+
+function setAlarm() {
+  // Get the total number of seconds
+  totalSeconds = parseInt(secondsInput.value);
+  // Clear any existing timers
+  clearInterval(countdown);
+  // Update the display with the initial time
+  displayTime(totalSeconds);
+
+  countdown = setInterval(() => {
+    totalSeconds--;
+    displayTime(totalSeconds);
+    if (totalSeconds === 0) {
+      clearInterval(countdown);
+      playAlarm();
+    }
+  }, 1000);
+  
+}
+
+function displayTime(seconds) {
+  // Calculate the number of minutes and seconds
+  const minutes = Math.floor(seconds / 60);
+  const remainderSeconds = seconds % 60;
+  // Add leading zeros to single digit numbers
+  const displayMinutes = `${minutes < 10 ? '0' : ''}${minutes}`;
```

Comment:
good style of coding! well done! 
i can give a small suggestion for "Leading Zeros". 
you could use `String.padStart()` method for shorter expression, 
example: 
```js
`${1}`.padStart(2, "0");
`${99}`.padStart(2, "0");
```

Reply:
one more thing. just in case if you wanna play with some more advanced techniques
check this out: 
[Temporal.Duration](https://tc39.es/proposal-temporal/docs/duration.html)
```js
duration = Temporal.Duration.from({
    minutes: 1,
    seconds: 5
});

while (duration.total("second") > 0) {
    duration = duration.subtract({seconds: 1});
    console.log(`${String(duration.minutes).padStart(2, "0")}:${String(duration.seconds).padStart(2, "0")}`);
}
```

Reply:
unfortunately this is still an experimental feature, 
but, you might google some other existing opensource up-to-date library that can be linked with your front-end.

Reply:
very interesting! Thank you! I need more time and exercises to explore .padStart and temporal.duration features. I will try it!